idea-plugin/src/main/java/com/palantir/gradle/jdks/PluginUpdateCheckerService.java

```diff
@@ -0,0 +1,77 @@
+ */
+
+package com.palantir.gradle.jdks;
+
+import com.intellij.externalDependencies.DependencyOnPlugin;
+import com.intellij.externalDependencies.ExternalDependenciesManager;
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManagerCore;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationGroupManager;
+import com.intellij.notification.NotificationType;
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.updateSettings.impl.pluginsAdvertisement.PluginsAdvertiser;
+import com.intellij.util.text.VersionComparatorUtil;
+import java.util.Optional;
+import java.util.Set;
+
+@Service(Service.Level.PROJECT)
+public final class PluginUpdateCheckerService {
+
+    private static final String PLUGIN_ID = "palantir-gradle-jdks";
+    private final Logger logger = Logger.getInstance(PluginUpdateCheckerService.class);
+    private final Project project;
+
+    public PluginUpdateCheckerService(Project project) {
+        this.project = project;
+    }
+
+    public void checkPluginVersion() {
+        PluginId pluginId = PluginId.getId(PLUGIN_ID);
+        IdeaPluginDescriptor pluginDescriptor = PluginManagerCore.getPlugin(pluginId);
+        if (pluginDescriptor == null) {
+            logger.info("Plugin " + pluginId + " not found");
+            return;
+        }
+        Optional<String> maybeMinVersion =
+                ExternalDependenciesManager.getInstance(project).getDependencies(DependencyOnPlugin.class).stream()
+                        .filter(dependencyOnPlugin ->
+                                dependencyOnPlugin.getPluginId().equals(pluginId.getIdString()))
+                        .map(DependencyOnPlugin::getMinVersion)
+                        .filter(version -> Optional.ofNullable(version).isPresent())
+                        .max(VersionComparatorUtil::compare);
+        boolean isPluginUpToDate = maybeMinVersion
+                .map(minVersion -> VersionComparatorUtil.compare(pluginDescriptor.getVersion(), minVersion) > 0)
+                .orElse(true);
+
+        if (!isPluginUpToDate) {
+            Notification notification = NotificationGroupManager.getInstance()
+                    .getNotificationGroup("Update Palantir plugins")
+                    .createNotification(
+                            "Update palantir-gradle-jdks plugin",
+                            String.format(
+                                    "Please update the plugin in the Settings window to a version higher than '%s'",
+                                    maybeMinVersion.get()),
+                            NotificationType.ERROR);
+            notification.notify(project);
+            PluginsAdvertiser.installAndEnablePlugins(Set.of(PLUGIN_ID), notification::expire);
+        }
+    }
+}
```

Comment:
Do we still get the other not so useful (see below) notification? Or is it suppressed somehow.

<img width="385" alt="Screenshot 2024-09-06 at 16 44 27" src="https://github.com/user-attachments/assets/17e43afc-1f98-4af3-afb0-8e8e895d79be">

Reply:
Yes, we still get it, we'd get both. I can remove my notification, but I thought it might be nice to actually tell you that the you need to use the pop-up window to update the plugin.